frontend/ui/screensaver.lua

```diff
@@ -147,4 +147,25 @@ function Screensaver:close()
     end
 end
 
+function Screensaver:isKindleSpecialOffers()
+    -- Look at the current blanket modules to see if the SO screensavers are enabled...
+    local lipc = require("liblipclua")
+    if not lipc then
+        logger.warn("could not load liblibclua")
+        return false
+    end
+    local lipc_handle = lipc.init("com.github.koreader.device")
+    if not lipc_handle then
+        logger.warn("could not get lipc handle")
+        return false
```

Comment:
I agree, we should return true in all the error handling code path, i.e. only show screen saver when we are absolutely sure it's safe to do so. For example, a new version of kindle firmware upgrade might cause the lipc call to fail on devices with special offers, in that case, we want to make sure the screen saver is still not displayed.